### PR TITLE
Azure key vault integration fix microsft api module

### PR DESCRIPTION
**Pull request by OfekMQmasters**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 1]
```diff
@@ -327,8 +335,8 @@ def _get_self_deployed_token_client_credentials(self, scope: Optional[str] = Non
         if self.scope or scope:
             data['scope'] = scope if scope else self.scope
 
-        if self.resource:
-            data['resource'] = self.resource
+        if self.resource or resource:
+            data['resource'] = resource if resource else self.resource  # type: ignore
```
Comment (Bargenish) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:339]:
```suggestion
            data['resource'] = resource or self.resource  # type: ignore
```

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 2]
```diff
@@ -305,15 +305,23 @@ def _get_self_deployed_token(self,
             return self._get_token_device_code(refresh_token, scope, integration_context)
         else:
             # by default, grant_type is CLIENT_CREDENTIALS
+            if self.multi_resource:
```
Comment (Bargenish) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:308]: Please add a unit test

**Review by Bargenish — COMMENTED** [review]
nice work
See my comments below

@guykeller need to add an alternative auth method (using device-code/add a XSOAR application).

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py (hunk 1 above)
Comment (OfekMQmasters) [reply Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py]: Done.

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py (hunk 2 above)
Comment (OfekMQmasters) [reply Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:308]: Added.

**Comment by OfekMQmasters** [comment]
1. I made the requested changes and added the relevant unit test. 
2. Regarding the device flow - in the first demo with Arseny and Meital we decided not to use the device flow as 
authorization in this integration, since the integration uses multiple scopes - https://vault.azure.net ( for keys, secrets and certificates commands) and https://management.azure.com for Key Vault commands, but the authorization with device flow can use only one of them. @Bargenish @guykeller

**Comment by bakatzir** [comment]
related: https://github.com/demisto/etc/issues/36247

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.py [hunk 3]
```diff
@@ -0,0 +1,1321 @@
...
+         client (KeyVaultClient):  Azure Key Vault API client.
+
+     Returns:
+         None
+     """
+    try:
+        client.ms_client.get_access_token(resource=MANAGEMENT_RESOURCE)
+        client.ms_client.get_access_token(resource=VAULT_RESOURCE)
+        client.list_key_vaults_request(1, 0)
+        # fetch_credentials(client,[''],[''],'xsoar-test-vault/test-sec-1')
```
[the remark this answers is not in the document]
Comment (OfekMQmasters) [reply Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.py:1121]: Here we verify:
1. That the client has access token for each resource.
2. Making an API request for listing key vaults (validating that a request does work).
In second thought, the first point should be enough. In addition, the commented out line should be deleted.
Let me know if anything else is missing and if do you approve my suggestion here.

**Review by guykeller — APPROVED** [review]

**Pull request merged**